src/components/radio/radio.ts

```diff
@@ -283,13 +275,30 @@ export class MdRadioButton implements OnInit {
 
     this.radioGroup = radioGroup;
 
+    if (this.radioGroup) {
+      this._listenToRadioGroup();
+    }
+
     radioDispatcher.listen((id: string, name: string) => {
       if (id != this.id && name == this.name) {
         this.checked = false;
       }
     });
   }
 
+  private _listenToRadioGroup() {
+    // name from `md-radio-group` overrides button name
+    this.name = this.radioGroup.name;
+    this.radioGroup.nameEmitter.subscribe((name: string) => {
+      this.name = name;
+    });
+
+    // listen to value changes on parent
+    this.radioGroup.change.subscribe((change: MdRadioChange) => {
+      this._checked = change.value == this.value;
```

Comment:
Replied to the previous comment - going to look into that as an option and see if it ends up being cleaner.